File: Data Structures & Algorithms/is-anagram/submission-6.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        count = {}

        for c1, c2 in zip(s,t):
            if c1 in count:
                count[c1]+=1
            else:
                count[c1]=1
            if c2 in count:
                count[c2]-=1
            else:
                count[c2]=-1

        return all(value == 0 for value in count.values())

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-6.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int l = 0, r = 0;

        unordered_set<char> window;
        int longest = 0;

        while (r < s.size()) {
            if (window.count(s[r])) {
                // invalidation condition
                window.erase(s[l]);
                l++;
                continue;
            } 
            window.insert(s[r]);
            longest = max(longest, r - l + 1);
            r++;
        }

        return longest;
    }
};
